oap-storage-mongo/src/main/java/oap/storage/mongo/OplogService.java

```diff
@@ -36,35 +36,44 @@
 import org.joda.time.DateTimeUtils;
 
 import java.io.Closeable;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 import static com.mongodb.client.model.Filters.and;
 import static com.mongodb.client.model.Filters.gt;
 import static com.mongodb.client.model.Filters.in;
 import static com.mongodb.client.model.Filters.regex;
 
+/**
+ * This service runs within a separate {@link Thread} and detects the changes on the Mongo collection via
+ * {@link OplogListener}
+ */
 @Slf4j
 public class OplogService implements Runnable, Closeable {
-    private final MongoClient mongoClient;
+    private final MongoClientWrapper mongoClient;
     private final ListMultimap<String, OplogListener> listeners = MultimapBuilder.hashKeys().arrayListValues().build();
     private MongoCursor<Document> cursor;
     private Thread thread;
+    private final AtomicBoolean running = new AtomicBoolean(false);
```

Comment:
Why atomic in this case? There is no concurrency requiring snapshot-like access. Start/stop are synced. Run is meant to read it asynchronously.